src/devices/imagedev/floppy.cpp

```diff
@@ -1781,8 +1783,10 @@ void floppy_sound_device::sound_stream_update(sound_stream &stream, std::vector<
 				sampleend = m_sample[idx].data.size();
 
 				// Mix it into the stream value
-				out += m_sample[idx].data[m_step_samplepos++];
-				if (m_step_samplepos >= sampleend)
+				m_step_samplepos++;
+				if (m_step_samplepos < sampleend)
+					out += m_sample[idx].data[m_step_samplepos];
+				else
```

Comment:
Again, this effectively replaces the postincrement of `m_step_samplepos` with a preincrement